pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -27,8 +35,13 @@
                        hostPlatform != stdenv.buildPlatform
 , extraMeta ? {}
 , hostPlatform
+
+# easy overrides to hostPlatform.platform members
+, autoModules ? hostPlatform.platform.kernelAutoModules
```

Comment:
@teto So I have a feeling I'm on the verge of spreading myself too thin, so I'm going to leave reviewing the actual kernel building code to @dezgeg and others.

Happy to comment on the general principle of config though. So *if* `hostPlatform.platform.kernelAutoModules`, etc, are used in multiple packages, we should probably keep them as and *not* allow per-package overrides. But in most (all?) of these cases, that's not the case. Then there's little point of keeping anything in `*Platform` at all.

So, if @dezgeg is OK with it, I'd be happy just making this `autoModules ? hostPlatform.isx86` is something. Or maybe there is a multiple-package-principle here of precise vs multiplatform configurations so we could have `autoModules ? !hostPlatform.preciseConfig`. Again, coordinate with @dezgeg and kernel people who are more up to speed on the specifics of this config me, but I just wanted to articulate the general principle of changes like this as I see it.

Reply:
I am not sure what motivates setting the autoModules to true, is it x86 only ?
I would favor changing this (if need be) in a follow up PR.

Reply:
The "multiplatform" examples in `lib.systems.examples` also do this, so I gather it boils down to trying to create a one-size-fits-all kernel vs tailor it to an exact machine.

Reply:
Well, it's not so clear-cut; it also affects whether you get a kernel that has support for all the possible WiFi dongles, PCI cards, etc.

I'd say keep the `platform.kernelAutoModules` for now since indeed this is a large patch already.

Reply:
Well conversely, could we just not add the extra parameters then? It looks like this PR does other things which are good/useful on their own.

Reply:
The initial motivation for the PR was to override these so I would hate it to see it go xD
I do a bit of kernel development and having to compile every single module is really a pain. This override cuts down compilation time a lot.

Reply:
@teto I'm not saying *never* do that :), but just do it in a second PR. I like the separation of concerns to better understand each is isolation, and (long term) I don't believe we should have override options for anything that is also in `*Platform`. [I also am confused why this would help with compilation times, but that is a separate matter.]

Reply:
having kernelAutoModules set to true mean sthat nix builds ton of modules I don't use, (GPU drivers etc) because it answers M to the interactive kernel config generator. Before your indications and even with it, I still find it hard to override the platform.kernelAutoModules value, having the override makes it more nixpkgs-like. It's no problem having it out-of-tree, it should generate less rebase conflicts as it's much smaller patch.